fake_tb.go

```diff
@@ -59,15 +58,17 @@ func newRecordingTB() *fakeTB {
 	}
 }
 
+func (tb *fakeTB) checkPanic() {
+	if r := recover(); r != nil {
+		tb.panicked = true
+		tb.recovered = r
+		tb.recoverCallers = getCallers()
+	}
+}
+
 func (tb *fakeTB) postTest() {
 	defer close(tb.completed)
 
-	defer func() {
-		if r := recover(); r != nil {
```

Comment:
this `defer` didn't catch panics previously for a couple of reasons:
 - the recover was inside of a nested `defer` (since `postTest` itself is run as a `defer`)
 - `recover` only works when the deferred function calls it directly.

E.g.,
if `recover` happens within a function call of the deferred function, it cannot recover the panic:
```go
func main() {
	defer func() {
		func() {
			if r := recover(); r != nil {
				fmt.Println("nested recover", r)
			}
		}()
	}()
	panic("foo")
}
```
https://go.dev/play/p/lq_fDfEPha9

While it works without the nested function:
```go
func main() {
	defer func() {
		if r := recover(); r != nil {
			fmt.Println("nested recover", r)
		}
	}()
	panic("foo")
}
```
https://go.dev/play/p/dwA1yoXm9ZV